tests/regression/test_aw.py

```diff
@@ -0,0 +1,120 @@
+from firedrake import *
+import pytest
+import numpy as np
+
+
+convergence_orders = lambda x: np.log2(np.array(x)[:-1] / np.array(x)[1:])
+
+
+@pytest.fixture(scope='module', params=["conforming", "nonconforming"])
+def stress_element(request):
+    if request.param == "conforming":
+        return FiniteElement("AWc", triangle, 3)
+    elif request.param == "nonconforming":
+        return FiniteElement("AWnc", triangle, 2)
+    else:
+        raise ValueError("Unknown family")
+
+
+def test_aw(stress_element):
+    N_base = 2
+    mesh = UnitSquareMesh(N_base, N_base)
+    mh = MeshHierarchy(mesh, 4)
+
+    V = FunctionSpace(mesh, mesh.coordinates.ufl_element())
+
+    # Warp the meshes
+    eps = Constant(1 / 2**(N_base-1))
+    x, y = SpatialCoordinate(mesh)
+    new = Function(V).interpolate(as_vector([x + eps*sin(2*pi*x)*sin(2*pi*y),
+                                             y - eps*sin(2*pi*x)*sin(2*pi*y)]))
+    coords = [new]
+    for mesh in mh[1:]:
+        fine = Function(mesh.coordinates.function_space())
+        prolong(new, fine)
+        coords.append(fine)
+        new = fine
+    for mesh, coord in zip(mh, coords):
+        mesh.coordinates.assign(coord)
+
+    nu = Constant(0.25)
+    lam = Constant(1)
+    mu = lam*(1 - 2*nu)/(2*nu)
+
+    I = Identity(2)
+
+    # Evaluation of a constant compliance tensor
+    # (in the homogeneous isotropic case)
+    def A(sig):
+        return (1/(2*mu))*(sig - nu*tr(sig)*I)
+
+    # Linearised strain rate tensor
+    def epsilon(u):
+        return sym(grad(u))
+
+    l2_u = []
+    l2_sigma = []
+    l2_div_sigma = []
+
+    element = MixedElement([stress_element, VectorElement("DG", mesh.ufl_cell(), 1)])
+    for msh in mh[1:]:
+        x, y = SpatialCoordinate(msh)
+        uex = as_vector([sin(pi*x)*sin(pi*y), sin(pi*x)*sin(pi*y)])
+        sigex = as_tensor([[cos(pi*x)*cos(3*pi*y), y + 2*cos(pi*x/2)],
+                           [y + 2*cos(pi*x/2), -sin(3*pi*x)*cos(2*pi*x)]])
+
+        # Compliance and constraint residuals of the MMS
+        comp_r = A(sigex) - epsilon(uex)
+        cons_r = div(sigex)
+
+        V = FunctionSpace(msh, element)
+
+        Uh = Function(V)
+
+        (sigh, uh) = split(Uh)
+        (tau, v) = TestFunctions(V)
+
+        n = FacetNormal(msh)
+
+        # Hellinger--Reissner residual, incorporating MMS residuals
+        F = (
+            + inner(A(sigh), tau)*dx
+            + inner(div(tau), uh)*dx
+            + inner(div(sigh), v)*dx
+            - inner(comp_r, tau)*dx
+            - inner(cons_r, v)*dx
+            - inner(dot(tau, n), uex)*ds
+            )  # noqa: E123
+
+        params = {"snes_type": "ksponly",
+                  "snes_monitor": None,
```

Comment:
with `ksponly`, you don't need a `snes_monitor` or `snes_converged_reason`.

Reply:
I agree you don't need it. I kept it so that if someone runs it themselves while requesting stdout, the system prints the residuals. By default the test harness swallows the output unless something goes wrong.

Reply:
You're right -- it's in a test and you get some information if it fails and you turn on the printout.  I was wondering if the potential reader would be confused by the options and not know the use case.

Reply:
I think these options are pretty common and the risk of confusion is minimal.